modules/videoio/include/opencv2/videoio.hpp

```diff
@@ -174,11 +174,54 @@ enum VideoCaptureProperties {
        CAP_PROP_CHANNEL       =43, //!< Video input or Channel Number (only for those cameras that support)
        CAP_PROP_AUTO_WB       =44, //!< enable/ disable auto white-balance
        CAP_PROP_WB_TEMPERATURE=45, //!< white-balance color temperature
+       CAP_PROP_INT_CODEC     =46, //!< mapping of internal capture codec to opencv RawCodec
+       CAP_PROP_INT_PX_FORMAT =47, //!< mapping of internal capture pixel format to opencv RawPixelFormat
 #ifndef CV_DOXYGEN
        CV__CAP_PROP_LATEST
 #endif
      };
 
+/** @brief %VideoContainer raw codecs
+*/
+enum RawCodec
+{
+    VideoCodec_MPEG1,
+    VideoCodec_MPEG2,
+    VideoCodec_MPEG4,
+    VideoCodec_VC1,
+    VideoCodec_H264,
+    VideoCodec_JPEG,
+    VideoCodec_HEVC,
+    VideoCodec_VP8,
+    VideoCodec_VP9,
+    VideoCodec_NumCodecs,
+
+    // Uncompressed YUV
+    VideoCodec_YUV420 = (('I' << 24) | ('Y' << 16) | ('U' << 8) | ('V')),   // Y,U,V (4:2:0)
```

Comment:
@mshabunin Those 

> Uncompressed YUV

are codecs returned by the nvcuvid video parser and can be removed, I incorrectly included them here.

I could use FOURCC codes for the remaining codecs but because there can be a many to one mapping from ffmpeg FOURCC codes to codec (h264, x264 etc.) it seemed more suitable to return the ffmpeg codec, do you agree?